### PR TITLE
feat: add utility function to get supported chains from the Security Alerts API

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

This PR aims to include a new utility function to get supported chains from the Security Alerts API when enabled.

#### Related Repository
Refer to the [Security Alerts API repository](https://github.com/consensys-vertical-apps/va-mmcx-security-alerts-api) for more details.

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1870

## **Manual testing steps**

1. Test blockaid regression

2. add the envs 
```shell
export SECURITY_ALERTS_API_URL='https://security-alerts.dev-api.cx.metamask.io'
export SECURITY_ALERTS_API_ENABLED='true'
```
- Go to test dapp and trigger on of the malicious signatures
- To verify in chrome go to dev tools > network. Search for `security-alerts` and find the call to the API service.

Existing PPOM logic should function as before, even with the above environment variables added, due to the fallback to the controller in the event of an error.

## **Screenshots/Recordings**

[Screencast from 19-07-2024 14:42:48.webm](https://github.com/user-attachments/assets/ba6c2f55-11e6-49f5-b9dc-120ed77390bd)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

**Comment by jpuri** [comment]
There is usage of `BLOCKAID_SUPPORTED_CHAIN_IDS` here also: https://github.com/MetaMask/metamask-mobile/blob/f7d8e61c4b4caab2623b208ad33a4db492700ad7/app/util/blockaid/index.ts#L20

File: app/lib/ppom/security-alerts-api.ts [hunk 1]
```diff
@@ -39,6 +45,20 @@ async function postRequest(endpoint: string, body: unknown) {
...
+
+  const response = await fetch(url);
+
+  if (!response.ok) {
+    throw new Error(
+      `Security alerts API request failed with status: ${response.status}`,
+    );
+  }
+
+  return response.json();
```
Comment (matthewwalsh0) [inline app/lib/ppom/security-alerts-api.ts:59]: Can we remove the duplication and create a `request` function for example that accepts the method and some extra properties?

File: app/lib/ppom/security-alerts-api.ts [hunk 2]
```diff
@@ -19,6 +21,10 @@ export async function validateWithSecurityAlertsAPI(
   return postRequest(endpoint, request);
 }
 
+export async function getSupportedChains(): Promise<Hex[]> {
```
Comment (matthewwalsh0) [inline app/lib/ppom/security-alerts-api.ts:24]: Maybe `getSecurityAlertsAPISupportedChainIds` to be explicit?

File: app/lib/ppom/ppom-util.ts [hunk 3]
```diff
@@ -119,6 +121,20 @@ async function validateRequest(req: PPOMRequest, transactionId?: string) {
   }
 }
 
+async function isChainSupported(chainId: Hex): Promise<boolean> {
+  try {
+    const supportedChains = isSecurityAlertsAPIEnabled()
+      ? await getSupportedChains()
+      : BLOCKAID_SUPPORTED_CHAIN_IDS;
+    return supportedChains.includes(chainId);
```
Comment (matthewwalsh0) [inline app/lib/ppom/ppom-util.ts:129]: Minor, but could we remove some duplication with something like:

```
let supportedChainIds = BLOCKAID_SUPPORTED_CHAIN_IDS;

try {
    if(isSecurityAlertsAPIEnabled()) {
        supportedChainIds = getSupportedChainds();
    }
} catch (e) {
    Logger.log(
      `Error fetching supported chains from security alerts API: ${e}`,
    );
}

return supportedChainIds.includes(chainId);
```

File: app/lib/ppom/ppom-util.test.ts [hunk 4]
```diff
@@ -238,6 +238,9 @@ describe('PPOM Utils', () => {
 
     it('uses security alerts API if enabled', async () => {
       isSecurityAlertsEnabledMock.mockReturnValue(true);
+      jest
```
Comment (matthewwalsh0) [inline app/lib/ppom/ppom-util.test.ts:241]: Minor, but could we create a persistent variable for this such as `getSupportedChainsMock` like we do for `isSecurityAlertsEnabledMock`?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: app/lib/ppom/security-alerts-api.ts (hunk 1 above)
Comment (vinistevam) [reply app/lib/ppom/security-alerts-api.ts]: Great suggestions, I created the `request` function with the same options supported by fetch.

File: app/lib/ppom/ppom-util.ts (hunk 3 above)
Comment (vinistevam) [reply app/lib/ppom/ppom-util.ts]: Nice! Pushed the suggestion. https://github.com/MetaMask/metamask-mobile/pull/10267/commits/ad145f490243d499ea835d9731af623d39b0b7ff

File: app/lib/ppom/ppom-util.test.ts (hunk 4 above)
Comment (vinistevam) [reply app/lib/ppom/ppom-util.test.ts]: Done. https://github.com/MetaMask/metamask-mobile/pull/10267/commits/ad145f490243d499ea835d9731af623d39b0b7ff

File: app/lib/ppom/ppom-util.ts [hunk 5]
```diff
@@ -119,6 +121,22 @@ async function validateRequest(req: PPOMRequest, transactionId?: string) {
   }
 }
 
+export async function isChainSupported(chainId: Hex): Promise<boolean> {
```
Comment (matthewwalsh0) [inline app/lib/ppom/ppom-util.ts:124]: Does this need to be exported as it's just part of the validate implementation?

Ideally we wouldn't expose but test indirectly?

**Review by matthewwalsh0 — DISMISSED** [review]

File: app/lib/ppom/ppom-util.ts (hunk 5 above)
Comment (vinistevam) [reply app/lib/ppom/ppom-util.ts]: Removed the `export` https://github.com/MetaMask/metamask-mobile/pull/10267/commits/b5b8db1ec62aa5914175a6ee7230e6b2c3ed11bd

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by vinistevam** [comment]
> There is usage of `BLOCKAID_SUPPORTED_CHAIN_IDS` here also:
> 
> https://github.com/MetaMask/metamask-mobile/blob/f7d8e61c4b4caab2623b208ad33a4db492700ad7/app/util/blockaid/index.ts#L20

I will open a ticket and the refactor will be handled in this [PR](https://github.com/MetaMask/metamask-mobile/pull/10518), cause it was getting too big.

**Pull request merged**